src/nl/hannahsten/texifyidea/settings/conventions/TexifyConventionsConfigurable.kt

```diff
@@ -0,0 +1,216 @@
+                                when (value.type!!) {
+                                    LabelConventionType.ENVIRONMENT -> TexifyIcons.DOT_ENVIRONMENT
+                                    LabelConventionType.COMMAND -> TexifyIcons.DOT_COMMAND
+                                }
+                            }
+                            else {
+                                null
+                            }
+
+                            return this
+                        }
+                    }
+                }
+            }
+
+        val enabledColumnInfo =
+            object : TableModelEditor.EditableColumnInfo<LabelConvention, Boolean>("Should have label") {
+                override fun getColumnClass(): Class<*> = Boolean::class.java
+                override fun getTooltipText(): String =
+                    "If enabled, TeXiFy issues a warning if the given Latex element does not have a label"
+
+                override fun valueOf(item: LabelConvention): Boolean = item.enabled
+                override fun setValue(item: LabelConvention, value: Boolean) {
+                    item.enabled = value
+                }
+
+                override fun getMaxStringValue(): String? {
+                    // Give a hint that the column won't ever need more space than its label
+                    return super.getName()
+                }
+            }
+
+        val itemEditor = object : CollectionItemEditor<LabelConvention> {
+            override fun getItemClass(): Class<out LabelConvention> = LabelConvention::class.java
+
+            override fun clone(item: LabelConvention, forInPlaceEditing: Boolean): LabelConvention {
+                TODO("Not yet implemented")
+            }
+
+            override fun isRemovable(item: LabelConvention): Boolean = false
+        }
+
+        labelConventionsTable = object : TableModelEditor<LabelConvention>(
+            listOf(nameColumnInfo, prefixColumnInfo, enabledColumnInfo).toTypedArray(),
+            itemEditor,
+            "Label Conventions"
+        ) {
+            override fun canCreateElement(): Boolean = false
+        }
+        labelConventionsTable.disableUpDownActions()
+        return labelConventionsTable
+    }
+
+    override fun createComponent(): JComponent {
+
+        schemesPanel = TexifyConventionsSchemesPanel(unsavedSettings)
+
+        /**
+         * Save the current scheme and load the new scheme whenever the combobox changes
+         */
+        schemesPanel.addListener(object : TexifyConventionsSchemesPanel.Listener {
+            override fun onCurrentSchemeWillChange(scheme: TexifyConventionsScheme) {
+                saveScheme(scheme)
+            }
+
+            override fun onCurrentSchemeHasChanged(scheme: TexifyConventionsScheme) {
+                loadScheme(scheme)
+            }
+        })
+
+        labelConventionsTable = createConventionsTable()
+
+        maxSectionSize = JBIntSpinner(4000, 100, Integer.MAX_VALUE)
```

Comment:
This spinner isn't yet working? It resets if I try to change it

Reply:
Ah, no, it was resetting because I was providing a value lower than the minimum of 100. But it only resets when you click ok so that's very confusing, maybe just leave the minimum at 1